core/templates/dev/head/pages/exploration_player/ExplorationPlayerStateService.js

```diff
@@ -28,24 +28,22 @@ oppia.factory('ExplorationPlayerStateService', [function() {
     },
 
     isInteractionInline: function(stateName) {
-      if (stateName === null) {
-        return true;
-      }
       return exploration.isInteractionInline(stateName);
     },
 
     getInteractionInstructions: function(stateName) {
-      if (stateName === null) {
-        return false;
-      }
       return exploration.getInteractionInstructions(stateName);
     },
 
     isStateTerminal: function(stateName) {
+      return exploration.isStateTerminal(stateName);
+    },
+
+    isStateShowingConceptCard: function(stateName) {
```

Comment:
Honestly, this is a hack. You're just passing null in, which is not a valid state name in the first place :)

That said, I think it's fine to merge this as long as we do a follow-up PR immediately after this one to introduce a Card object and move some of these methods to that object, so as to avoid this PR becoming too big. (Note that the Card object wouldn't have a state name property.) Would that be OK?

Reply:
Yeah, that would be fine. So, the transcript would then store 2 fields for stateName and the card object?

Reply:
I think the transcript can store:

- isConceptCard: bool
- stateName: string or null (null only if it's concept card)
- card: card object, should be the same format regardless of whether it's concept card or not

Please file an issue for this with all the necessary details before merging this PR. Thanks!

Reply:
Done, created [issue](https://github.com/oppia/oppia/issues/5329).

Reply:
Actually, whether it's conceptCard can be checked with stateName, so is a separate bool field required?

Reply:
Ok, sg!